src/Compilers/Core/Portable/Generated/Operations.Generated.cs

```diff
@@ -9047,6 +9054,11 @@ internal sealed partial class OperationCloner : OperationVisitor<object?, IOpera
         public override IOperation DefaultVisit(IOperation operation, object? argument) => throw ExceptionUtilities.Unreachable;
         private ImmutableArray<T> VisitArray<T>(ImmutableArray<T> nodes) where T : IOperation => nodes.SelectAsArray((n, @this) => @this.Visit(n), this);
         private ImmutableArray<(ISymbol, T)> VisitArray<T>(ImmutableArray<(ISymbol, T)> nodes) where T : IOperation => nodes.SelectAsArray((n, @this) => (n.Item1, @this.Visit(n.Item2)), this);
+        public override IOperation VisitEmpty(IEmptyOperation operation, object? argument)
+        {
+            var internalOperation = (EmptyOperation)operation;
+            return new EmptyOperation(internalOperation.OwningSemanticModel, internalOperation.Syntax, internalOperation.IsImplicit);
```

Comment:
All of this information is available directly on `IEmptyOperation`, why do the cast here? This seems to be unnecessary and imply there is only one valid implementation of `IEmptyOperation`.

Reply:
`OwningSemanticModel` is not directly available on `IEmptyOperation`.

Reply:
> imply there is only one valid implementation of IEmptyOperation.

Also yes, there is only one valid implementation of IEmptyOperation.